starter_code/elevator.js

```diff
@@ -1,19 +1,44 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor () {
+    this.floor = 0
+    this.MAXFLOOR = 10
+    this.requests = []
+    this.direction = 'up'
+    this.timeoutId = this.timeoutId
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start () {
+    this.timeoutId = setInterval(() => this.floorUp, 1000)
+  }
+  stop () {
+    clearInterval(this.start())
+  }
+  update () {
+    this.log()
+  }
+  _passengersEnter () { }
+  _passengersLeave () { }
+  floorUp () {
+    if (this.floor < this.MAXFLOOR) {
+      return this.floor++
+    }
+  }
+  floorDown () {
+    if (this.floor < 0) {
+      return this.floor--
+    }
+  }
+  call () {
+
+  }
+  log () {
+    console.log(` Direction: ${this.direction} | Floor: ${this.floor}`)
+  }
 }
 
-module.exports = Elevator;
+module.exports = Elevator
+
+let elevator = new Elevator()
```

Comment:
this should be const, cause you will not reasign it

Reply:
This should be declared in index file. Elevator and person files should be just for the constructor and methods